src/librbd/io/ObjectRequest.cc

```diff
@@ -246,6 +253,22 @@ void ObjectReadRequest<I>::handle_read_object(int r) {
     return;
   }
 
+  // merge ExtentResults to a single sparse bufferlist
+  int pos = 0;
+  for (auto& [read_data, extent_map] : m_extent_results) {
+    auto read_data_ptr = read_data.get();
+    auto extent_map_ptr = extent_map.get();
+    m_read_data->append(*read_data_ptr);
+    if (extent_map_ptr->size() == 0) {
+      m_extent_map->push_back(std::make_pair(m_extents[pos].first,
+                                             read_data_ptr->length()));
+    } else {
+      m_extent_map->insert(m_extent_map->end(),
```

Comment:
Nit: might be good to assert that the extent map total length actually matches the buffer length